### PR TITLE
Use nullptr instead of NULL for all files in src/

### PR DESCRIPTION
**Pull request by sickpig**
The substitution has not been carried out for leveldb, secp256k1, univalue and ctaes subtrees.

The changes has been made in an scripted fashion using:

    grep -Iwr NULL src/ | grep -v univalue | grep -v secp256k1 | grep -v leveldb | grep -v ctaes | awk -F ':' '{print $1}' | sort | uniq | xargs -I {} sed --in-place -e 's/\<NULL\>/nullptr/g' {}

### CODE REVIEW

File: src/torcontrol.cpp [hunk 1]
```diff
@@ -629,15 +629,15 @@ void TorController::protocolinfo_cb(TorControlConnection &_conn, const TorContro
...
         }
-        // Prefer NULL, otherwise SAFECOOKIE. If a password is provided, use HASHEDPASSWORD
+        // Prefer nullptr, otherwise SAFECOOKIE. If a password is provided, use HASHEDPASSWORD
         /* Authentication:
          *   cookie:   hex-encoded ~/.tor/control_auth_cookie
          *   password: "password"
          */
         std::string torpassword = GetArg("-torpassword", "");
-        if (methods.count("NULL"))
+        if (methods.count("nullptr"))
```
Comment (gandrewstone) [inline src/torcontrol.cpp]: not a good substitution

File: src/torcontrol.cpp [hunk 2]
```diff
@@ -629,15 +629,15 @@ void TorController::protocolinfo_cb(TorControlConnection &_conn, const TorContro
...
         {
-            LOG(TOR, "tor: Using NULL authentication\n");
+            LOG(TOR, "tor: Using nullptr authentication\n");
```
Comment (gandrewstone) [inline src/torcontrol.cpp]: keep NULL

Comment (sickpig) [reply src/torcontrol.cpp]: will do

File: src/torcontrol.cpp (hunk 1 above)
Comment (sickpig) [reply src/torcontrol.cpp]: good catch. fixing.

**Comment by sickpig** [comment]
rebased.

**Pull request merged**